tutorials/cloud-spanner-aspnet/index.md

```diff
@@ -0,0 +1,107 @@
+        }
+    }
+
+
+When you run the application, use a utility to send different REST requests to test your API.  I like using [Postman](https://chrome.google.com/webstore/detail/postman/fhbjgbiflinjbdggehcddcbncdddomop), which is a Chrome extension but you can use something else like curl.  Use Postman to add a new book to the database as below.  Note: replace the port number to match your server.  So, the ‘56777’ in [http://localhost:56777](http://localhost:56777) should be replaced with the port your local server is running on.
+
+![image alt text](image_3.png)
+
+### **Getting Books.**
+
+Now that we have code to insert records, we can add more code to list all books or a single book.
+
+Add the following methods to your controller.
+
+    . . .
+    [HttpGet]
    public async Task<IActionResult> GetAll()
    {
        var result = new List<Book>();
     
        using (var connection = new SpannerConnection(
            $"Data Source=projects/{_myProject}/instances/myspanner/databases/books"))
        {
            var selectCmd = connection.CreateSelectCommand("SELECT * FROM bookTable");
            using (var reader = await selectCmd.ExecuteReaderAsync())
            {
                while (await reader.ReadAsync())
                {
                    result.Add(new Book
                    {
                        Id = reader.GetFieldValue<string>("ID"),
                        Title = reader.GetFieldValue<string>("Title"),
                        Author = reader.GetFieldValue<string>("Author"),
                        PublishDate = reader.GetFieldValue<DateTime>("PublishDate")
                    });
                }
            }
        }
     
        return Ok(result);
    }
     
    [HttpGet]
    [Route("{id}", Name="GetBookById")]
    public async Task<IActionResult> Get(string id)
    {
        using (var connection = new SpannerConnection(
            $"Data Source=projects/{_myProject}/instances/myspanner/databases/books"))
        {
            var selectCommand =connection.CreateSelectCommand($"SELECT * FROM bookTable WHERE ID='{id}'");
            using (var reader = await selectCommand.ExecuteReaderAsync())
            {
                while (await reader.ReadAsync())
                {
                    return Ok(new Book
                    {
                        Id = reader.GetFieldValue<string>("ID"),
                        Title = reader.GetFieldValue<string>("Title"),
                        Author = reader.GetFieldValue<string>("Author"),
                        PublishDate = reader.GetFieldValue<DateTime>("PublishDate")
                    });
                }
            }
        }
     
        return NotFound();
    }
+    . . .
+
+
+Now, run your program.  Using Postman, perform an HTTP GET on [http://localhost:56777/api/books](http://localhost:56777/api/books) to return all books or [http://localhost:56777/api/books/[ID](http://localhost:56777/api/books/[ID)] to retrieve a single book.
+
```

Comment:
Suggest new section that combines HTTP POST/GET info.

## Adding books and testing the application

To run the application, use a utility to send REST requests to test the API. This example shows [Postman](https://chrome.google.com/webstore/detail/postman/fhbjgbiflinjbdggehcddcbncdddomop), a Chrome extension, but you can use curl or another utility. 

1.  To add a book to the database, perform an `HTTP POST` with the port that your local server is running on. For example, replace port *56777* in <code>http://localhost:56777</code>. 
1.  To test the application, perform an HTTP GET.
    *   To return all books, enter:  <code>http://localhost:<em>PORT_NUMBER</em>/api/books</code>
    *   To return a single book, enter:  <code>http://localhost:<em>PORT_NUMBER</em>/api/books/<em>ID</em></code>